src/pipeline/nodes/VirtualNode.groovy

```diff
@@ -111,6 +111,60 @@ def stopFetchLogs(String library_dir, String target_dir){
     }
 }
 
+def remoteStartFetchLogs(String target_dir, Map remote_info, String library_dir=null){
+    try{
+       if(library_dir == null)
+       {
+         library_dir = "$WORKSPACE/on-build-config"
+         new pipeline.common.ShareMethod().checkoutOnBuildConfig(library_dir)
+       }
+        withCredentials([
+             usernamePassword(credentialsId: 'BMC_VNODE_CREDS',
+                             passwordVariable: 'BMC_VNODE_PASSWORD',
+                             usernameVariable: 'BMC_VNODE_USER')
+        ]) {
+              def remote_ip = remote_info["ip"]
+              def remote_user = remote_info["username"]
+              def remote_pass = remote_info["password"]
+              dir(target_dir){
+                sh """#!/bin/bash -ex
+                  current_dir=`pwd`
+                  pushd $library_dir/src/pipeline/nodes/ansible
+                    echo "ova-post-test ansible_host=$remote_ip ansible_user=$remote_user ansible_ssh_pass=$remote_pass ansible_become_pass=$remote_pass" > hosts
+                    ansible-playbook -i hosts main.yml --tags "start" --extra-vars "library_dir=$library_dir target_dir=\$current_dir BMC_CRED=$BMC_VNODE_USER:$BMC_VNODE_PASSWORD"
+                  popd
+                """
+              }
+        }
+    } catch(error){
+        echo "[WARNING] Failed to fetch logs of virtual nodes with error: $error"
+    }
+}
+
+def remoteStopFetchLogs(String target_dir, Map remote_info, String library_dir=null){
+    try{
+        if(library_dir == null)
+        {
+          library_dir = "$WORKSPACE/on-build-config"
+          new pipeline.common.ShareMethod().checkoutOnBuildConfig(library_dir)
+        }
+        def remote_ip = remote_info["ip"]
+        def remote_user = remote_info["username"]
+        def remote_pass = remote_info["password"]
+        dir(target_dir){
+          sh """#!/bin/bash -ex
+            current_dir=`pwd`"/"
```

Comment:
same, why add "/"

Reply:
same

Reply:
If ```remoteStartFetchLogs``` is ok without ```pwd"/"```, there is no need to add "/", right?

Reply:
right~